### PR TITLE
some fixes for windows x64 support

### PR DESCRIPTION
**Pull request by zeliard**
1. add null pointer checks
2. fix wrong memset argument
3. fix wrong pointer castings (64bit pointer to 32bit integers)

### CODE REVIEW

**Comment by ghost** [comment]
Hi **@zeliard**, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

**Comment by ghost** [comment]
**@zeliard**, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSOTBOT;

File: src/replication.c [hunk 1]
```diff
@@ -1970,7 +1970,7 @@ void replicationCron(void) {
     }
 
     /* Timed out master when we are an already connected slave? */
-    if (server.masterhost && server.repl_state == REDIS_REPL_CONNECTED &&
+	if (server.masterhost && server.repl_state == REDIS_REPL_CONNECTED && server.master &&
```
Comment (orangemocha) [inline src/replication.c:1973]: This looks like an issue that's not specific to Windows. Could you elaborate on what the bug is? Would it make sense to submit this to antirez\redis?

File: src/sentinel.c [hunk 2]
```diff
@@ -582,7 +582,7 @@ void sentinelEvent(int level, char *type, sentinelRedisInstance *ri,
     robj *channel, *payload;
 
     /* Handle %@ */
-    if (fmt[0] == '%' && fmt[1] == '@') {
+	if (fmt[0] == '%' && fmt[1] == '@' && ri != NULL) {
```
Comment (orangemocha) [inline src/sentinel.c:585]: This looks like an issue that's not specific to Windows. Could you elaborate on what the bug is? Would it make sense to submit this to antirez\redis?

File: src/sentinel.c [hunk 3]
```diff
@@ -2123,9 +2123,9 @@ void sentinelRefreshInstanceInfo(sentinelRedisInstance *ri, const char *info) {
 
     /* Handle slaves replicating to a different master address. */
     if ((ri->flags & SRI_SLAVE) &&
-        role == SRI_SLAVE &&
+		role == SRI_SLAVE &&
         (ri->slave_master_port != ri->master->addr->port ||
-         strcasecmp(ri->slave_master_host,ri->master->addr->ip)))
+		(ri->slave_master_host && strcasecmp(ri->slave_master_host, ri->master->addr->ip))))
```
Comment (orangemocha) [inline src/sentinel.c:2128]: This looks like an issue that's not specific to Windows. Could you elaborate on what the bug is? Would it make sense to submit this to antirez\redis?

**Comment by orangemocha** [comment]
Thank you so much @zeliard , these are some great catches! :+1: 

I pointed out a few changes that don't seem Windows-specific. I would like to understand how critical those issues are. If they are not critical, and they affect antirez\redis as well, it would be better to submit them there.

All the other ones are legitimate bugs. The conversions from void \* to long are technically portability bugs in antirez\redis, but we will take those fixes here urgently.

File: src/replication.c (hunk 1 above)
Comment (zeliard) [inline src/replication.c:1973]: Yes, you are right. This is not Windows-specific. 
The changes belonging to this commit ( ea5e668bc11a332d7084a5be5f421838450e8cbd ) is  not specific to Windows, too. 
Just added null-pointer check in order to remove potential memory-safety risk factors.

**Comment by zeliard** [comment]
Thanks @orangemocha 
But, it may be that there are more wrong pointer casting bugs.

Redis was made for *nix (linux, bsd, osx, ...) platforms which support LP64 models at first.
On those platforms regarding long as 64bit was no problem. But Windows on x64 uses LLP64 model, so we should check all "long" type variables when porting redis from *nix to Windows.

**Comment by orangemocha** [comment]
These fixes where incorporated in the 2.8.21 release. They prompted us to fix many portability issues with the usage of long. Thanks again, @zeliard !

**Pull request closed without merging**